rust/src/storage/file/mod.rs

```diff
@@ -101,17 +102,26 @@ impl StorageBackend for FileStorageBackend {
         if let Some(parent) = Path::new(path).parent() {
             fs::create_dir_all(parent).await?;
         }
+        let tmp_path = &format!("{}_{}", path, Uuid::new_v4().to_string());
         let mut f = fs::OpenOptions::new()
             .create(true)
             .truncate(true)
             .write(true)
-            .open(path)
+            .open(tmp_path)
             .await?;
 
         f.write_all(obj_bytes).await?;
         f.flush().await?;
-
-        Ok(())
+        drop(f);
```

Comment:
@houqp Does the `f.flush()` that you've added makes sense now? Also, is it actually required to flush a handle since it's being closed immediately

Reply:
yes, this is still needed, the content could still be partially cached in kernel without an explicit flush/sync.

after I took a second look at the doc, I think we should be using sync_all instead of flush.

Reply:
Filed https://github.com/delta-io/delta-rs/issues/368 as follow up.

Reply:
I've also changed flush to sync_all.